include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,179 @@
+            if( LL(i) > UL(i) ){
+                std::cout << "Invalid integration limits\n";
+                return false;
+            }
+        }
+        return true;
+    }else{
+        std::cout << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Hyper-Rectangle volume calculator in n-dimensions
+NT hyperRectVolume(VT LL, VT UL){
+    NT product=1;
+    if(legitLimits(LL,UL)){
+        for(int i=0; i<LL.rows(); ++i){
+            product = product * ( UL(i) - LL(i) );
+        }
+        return product;
+    }
+    else return 0;
+}
+
+// To sample a point between two n-dimensional points
+VT samplerBWLimits(VT LL, VT UL){
+    VT sample_point(LL.rows()); NT x;
+    for(int i=0; i<LL.rows(); ++i){
+            sample_point(i) = LL(i) + (NT)(rand()) / ((NT)(RAND_MAX/(UL(i) - LL(i))));     
+    }
+    return sample_point;
+}
+
+// Simple MC Integration over Hyper-Rectangles
+template < typename Functor >
+void SimpleMCIntegrate(Functor &Fx, Uint N ,VT LL, VT UL){
+    NT sum = 0;
+    if(legitLimits(LL,UL)){
+        for (int i = 1; i < N; i++ ) {
+            sum = sum + Fx(samplerBWLimits(LL,UL));
+        }    
+        std::cout << "Integral Value: " << hyperRectVolume(LL,UL) * sum / N << "\n";
+    }else{
+        std::cout << "Invalid integration limits\n";
+    }
+}
+
+template < typename Functor , typename Polytope >
+void SimpleMCPolytopeIntegrate(Functor &Fx, Polytope &P, Uint N,volumeType vType=CB,VT newOrigin=Origin){
+
+    // Polytope volumetric calculation params
+    Uint dim = P.dimension();
+    int walk_length = 10 + dim/10; 
+    NT e=0.1;
+    NT volume=0;
+
+    // Volume calculation for HPolytope
+    switch(vType)
+    {
+    case CB:     
+        volume = volume_cooling_balls<BallWalk, RNGType, HPOLYTOPE>(P, e, walk_length).second;
+        break;
+    case CG: 
+        volume = volume_cooling_gaussians<GaussianBallWalk, RNGType>(P, e, walk_length);
+        break;
+    case SOB: 
+        volume = volume_sequence_of_balls<BallWalk, RNGType>(P, e, walk_length);
+        break;    
+    default:  
+        volume = 0;
+        std::cout << "Error in enum:vType\n";
+        break;
+    }
+    
+    // For implementation of ReHMC walks
+    std::pair<Point, NT> inner_ball = P.ComputeInnerBall();
+    RNGType rng(1);
+    Point x0 = inner_ball.first;
+    NT r = inner_ball.second;
+    GaussianFunctor::parameters<NT, Point> params(x0, 2 / (r * r), NT(-1));
+    GaussianRDHRWalk::Walk<HPOLYTOPE, RNGType> walk(P, x0, params.L, rng);
+    NegativeGradientFunctor F(params);
+    NegativeLogprobFunctor f(params);
+    HamiltonianMonteCarloWalk::parameters<NT, NegativeGradientFunctor> hmc_params(F, dim);
+    HamiltonianMonteCarloWalk::Walk
+    <Point,HPOLYTOPE, RNGType, NegativeGradientFunctor, NegativeLogprobFunctor, Solver>
+    hmc( &P , x0, F, f, hmc_params);
+
+    // Check if origin is shifted
+    const bool &shiftedOrigin = ( newOrigin.rows()==0 )? false : true ;
```

Comment:
For applying shift to the sampled polytope points. it is `false` if no new point is provided default is taken as origin because `generate_cube()` generates the HPolytope around the origin.

Reply:
You should use the chebyshev center in this case. You can find it using `P.inner_ball()`.